app/lib/server/functions/createDirectRoom.js

```diff
@@ -90,13 +91,14 @@ export const createDirectRoom = function(members, roomExtraData = {}, options =
 	} else {
 		members.forEach((member) => {
 			const otherMembers = sortedMembers.filter(({ _id }) => _id !== member._id);
+			const memberWithPreferences = Users.findOneByUsername(member.username, { fields: { username: 1, 'settings.preferences': 1 } });
```

Comment:
instead of executing a find for every member, I'd say to execute a single find for all members and loop it result.